syntax = "proto3";

package helloworld;

message HelloRequest {
    string name = 1;
}

message HelloReply {
    string message = 1;
}

service Greeter {
    // 简单 RPC
    rpc SayHello (HelloRequest) returns (HelloReply);

    // 服务器端流式 RPC
    rpc SayHelloServerStreaming (HelloRequest) returns (stream HelloReply);

    // 客户端流式 RPC
    rpc SayHelloClientStreaming (stream HelloRequest) returns (HelloReply);

    // 双向流式 RPC
    rpc SayHelloBiDiStreaming (stream HelloRequest) returns (stream HelloReply);
}
